lib/browser.rb

```diff
@@ -89,156 +86,40 @@ class Browser
-  def name
-    NAMES[id]
-  end
-
-  # Get the browser identifier.
-  def id
-    NAMES.keys
-      .find {|id| respond_to?("#{id}?", true) ? send("#{id}?") : id }
-  end
-
-  # Return major version.
-  def version
-    if ie?
-      ie_version
-    else
-      full_version.to_s.split(".").first
-    end
-  end
-
-  # Return the full version.
-  def full_version
-    if ie?
-      ie_full_version
-    else
-      _, *v = *ua.match(VERSIONS.fetch(id, VERSIONS[:default]))
-      v.compact.first || "0.0"
-    end
-  end
-
-  # Return true if browser is modern (Webkit, Firefox 17+, IE9+, Opera 12+).
-  def modern?
-    self.class.modern_rules.any? {|rule| rule === self }
-  end
-
-  # Detect if browser is WebKit-based.
-  def webkit?
-    ua =~ /AppleWebKit/i && !edge?
-  end
-
-  # Detect if browser is QuickTime
-  def quicktime?
-    !!(ua =~ /QuickTime/i)
-  end
-
-  # Detect if browser is Apple CoreMedia.
-  def core_media?
-    !!(ua =~ /CoreMedia/)
-  end
-
-  # Detect if browser is PhantomJS
-  def phantom_js?
-    !!(ua =~ /PhantomJS/)
-  end
-
-  # Detect if browser is Safari.
-  def safari?
-    !!((ua =~ /Safari/) && ua !~ /Android|Chrome|CriOS|PhantomJS/)
-  end
-
-  def safari_webapp_mode?
-    !!((ipad? || iphone?) && ua =~ /AppleWebKit/)
-  end
-
-  # Detect if browser is Firefox.
-  def firefox?
-    !!(ua =~ /Firefox/)
-  end
-
-  # Detect if browser is Chrome.
-  def chrome?
-    !!(ua =~ /Chrome|CriOS/ && !opera? && !edge?)
-  end
-
-  # Detect if browser is Opera.
-  def opera?
-    !!(ua =~ /(Opera|OPR)/)
-  end
-
-  # Detect if browser is Silk.
-  def silk?
-    !!(ua =~ /Silk/)
-  end
-
-  # Detect if browser is Yandex.
-  def yandex?
-    !!(ua =~ /YaBrowser/)
-  end
-
-  # Detect if browser is UCBrowser.
-  def uc_browser?
-    !!(ua =~ /UCBrowser/)
-  end
-
-  def known?
-    id != :other
-  end
-
-  # Return a meta info about this browser.
-  def meta
-    Meta.constants.each_with_object(Set.new) do |meta_name, meta|
-      meta_class = Meta.const_get(meta_name)
-      meta.merge(meta_class.new(self).to_a)
-    end.to_a
-  end
-
-  alias_method :to_a, :meta
-
-  # Return meta representation as string.
-  def to_s
-    meta.to_a.join(" ")
+  modern_rules.tap do |rules|
+    rules << -> (b) { b.webkit? }
+    rules << -> (b) { b.firefox? && b.version.to_i >= 17 }
+    rules << -> (b) { b.ie? && b.version.to_i >= 9 && !b.compatibility_view? }
+    rules << -> (b) { b.edge? && !b.compatibility_view? }
+    rules << -> (b) { b.opera? && b.version.to_i >= 12 }
+    rules << -> (b) { b.firefox? && b.device.tablet? && b.platform.android? && b.version.to_i >= 14 } # rubocop:disable Metrics/LineLength
+  end
+
+  def self.new(user_agent, accept_language = nil)
+    [
+      Nokia,
+      UCBrowser,
+      PhantomJS,
+      BlackBerry,
+      Opera,
+      Edge,
+      InternetExplorer,
+      Firefox,
+      Chrome,
+      Safari,
+      Generic,
+    ].map {|klass| klass.new(user_agent, accept_language) }
+     .find(&:match?)
```

Comment:
Prefer `detect` over `find`.<br>Use 2 (not 1) spaces for indenting an expression spanning multiple lines.